playbooks/roles/provision_aws/tasks/ec2_teardown.yml

```diff
@@ -19,6 +20,7 @@
 - name: Delete EC2 instances
   ec2_instance:
     region: "{{ aws_region }}"
+    availability_zone: "{{ aws_zone }}"
```

Comment:
You shouldn't need to limit by availability zone here as it should still find the EC2 instances to delete. You can remove it.